shared-src/gradle-plugin/protokt/v1/gradle/ProtobufBuild.kt

```diff
@@ -52,51 +53,90 @@ internal fun configureProtobufPlugin(
-                        project.afterEvaluate {
-                            option("$KOTLIN_EXTRA_CLASSPATH=${extraClasspath(project, task)}")
-                            option("$GENERATE_TYPES=${ext.generate.types}")
-                            option("$GENERATE_DESCRIPTORS=${ext.generate.descriptors}")
-                            option("$GENERATE_GRPC_DESCRIPTORS=${ext.generate.grpcDescriptors}")
-                            option("$GENERATE_GRPC_KOTLIN_STUBS=${ext.generate.grpcKotlinStubs}")
-                            option("$FORMAT_OUTPUT=${ext.formatOutput}")
-                            option("$KOTLIN_TARGET=$target")
-                        }
+                        option("$KOTLIN_EXTRA_CLASSPATH=${extraClasspath(project, extensions)}")
+                        option("$GENERATE_TYPES=${ext.generate.types}")
+                        option("$GENERATE_DESCRIPTORS=${ext.generate.descriptors}")
+                        option("$GENERATE_GRPC_DESCRIPTORS=${ext.generate.grpcDescriptors}")
+                        option("$GENERATE_GRPC_KOTLIN_STUBS=${ext.generate.grpcKotlinStubs}")
+                        option("$FORMAT_OUTPUT=${ext.formatOutput}")
+                        option("$KOTLIN_TARGET=$target")
                     }
                 }
             }
+
+            project.handleExtraInputFiles(mainExtractProtoAdditions, testExtractProtoAdditions)
         }
     }
 }
 
-private fun extraClasspath(project: Project, task: GenerateProtoTask): String {
-    var extensions: FileCollection = project.configurations.getByName(EXTENSIONS)
-
-    if (task.isTest) {
-        extensions += project.configurations.getByName(TEST_EXTENSIONS)
+private fun Project.handleExtraInputFiles(main: List<TaskInputFiles>, test: List<TaskInputFiles?>) {
+    afterEvaluate {
+        handleExtractProtoAdditions(main, false)
+        handleExtractProtoAdditions(test, true)
     }
+}
 
-    // Must explicitly register input files here; if any extensions dependencies are project dependencies then Gradle
-    // won't pick them up as dependencies unless we do this. There may be a better way to do this but for now just
-    // manually do what protobuf-gradle-plugin used to do.
-    // https://github.com/google/protobuf-gradle-plugin/commit/0521fe707ccedee7a0b4ce0fb88409eefb04e59d
-    project.tasks.withType<ProtobufExtract> {
-        if (name.startsWith("extractInclude")) {
-            inputFiles.from(extensions)
+private fun Project.handleExtractProtoAdditions(additions: List<TaskInputFiles?>, test: Boolean) {
+    additions.filterNotNull().forEach {
+        // there is more than one task in multiplatform projects: e.g. extractIncludeProto and extractIncludeJvmMainProto
+        extractIncludeProtoTasks().forEach { task ->
+            val isTest = task.isTestTask()
+            if ((!test && !isTest) || (test && isTest)) {
+                logger.log(DEBUG_LOG_LEVEL, "Adding input files to task ${task.name} from ${it.taskName}")
```

Comment:
logger.debug?

Reply:
This constant maps to debug right now, but I like to be able to change the value to make logs noisier at will. I can back out that pattern in a separate PR but don't want to right now.